### PR TITLE
Better NVD parsing

### LINKED ISSUE #448: Use Jackson Streaming API for parsing NVD entries
During work on the PoC (#423) and the new ArtifactVersionVulnerabilityScore I felt that the `com.sap.oss.phosphor.fosstars.nvd.NVD#parse()` method consume a lot memory (~1.25GB heap) for parsing the NVD entries.

Recommendation would be to use the Jackson Streaming API not only for getting the `CVE_Items` array but also for parsing of each entry.
From my understanding this `NvdEntry entry : Json.mapper().readValue(parser, NvdEntry[].class)` parse the whole array at once with the object mapper.

### PR DESCRIPTION
**Pull request by artem-smotrakov**
Main updates:

- Remove `UnpatchedVulnerabilities.isUnpatched()` since it is not reliable.
- Updated `CpeMatch` not to store unused data and parse CPEs only once.
- Removed public `NVD.parse()` method. Instead, `NVD.search()` now parses data.
- Added `NVD.preload()` to load NVD entries in advance. The CLI preloads NVD entries if `--config` option is set.

In #448, it is said that currently ~1.25GB heap is consumed. After the update, a CLI run consumes ~780MB.

Closes #448

### CODE REVIEW

File: src/main/java/com/sap/oss/phosphor/fosstars/nvd/NVD.java [hunk 1]
```diff
@@ -196,30 +199,26 @@ boolean downloadFailed() {
...
-        if (matcher.match(entry)) {
-          result.add(entry);
+        if (matcher.match(nvdEntry)) {
+          result.add(nvdEntry);
         }
       }
+    };
+
+    if (!nvdEntries.isEmpty()) {
+      nvdEntries.forEach(collector);
```
Comment (sourabhsparkala) [inline src/main/java/com/sap/oss/phosphor/fosstars/nvd/NVD.java:214]: What is forEach doing here?

File: src/main/java/com/sap/oss/phosphor/fosstars/nvd/data/CpeMatch.java [hunk 2]
```diff
@@ -1,98 +1,67 @@
...
  * CPE match string or range.
  */
 @JsonInclude(JsonInclude.Include.NON_NULL)
 @JsonPropertyOrder({
-    "vulnerable",
     "cpe22Uri",
     "cpe23Uri",
-    "versionStartExcluding",
-    "versionStartIncluding",
     "versionEndExcluding",
```
Comment (sourabhsparkala) [inline src/main/java/com/sap/oss/phosphor/fosstars/nvd/data/CpeMatch.java:18]: I think this should be "versionStartIncluding"

File: src/main/java/com/sap/oss/phosphor/fosstars/nvd/data/CpeMatch.java [hunk 3]
```diff
@@ -1,98 +1,67 @@
...
+    "versionEndIncluding"
+})
+// the properties below are ignored because they are not used
+// that saves a bit of memory
+// when they become necessary, then can be enabled
+@JsonIgnoreProperties({
+    "vulnerable",
+    "cpe_name",
+    "versionStartExcluding",
+    "versionStartIncluding"
```
Comment (sourabhsparkala) [inline src/main/java/com/sap/oss/phosphor/fosstars/nvd/data/CpeMatch.java:28]: This should be "versionEndExcluding"

**Review by sourabhsparkala — COMMENTED** [review]
Few minor comments

File: src/main/java/com/sap/oss/phosphor/fosstars/nvd/NVD.java (hunk 1 above)
Comment (artem-smotrakov) [reply src/main/java/com/sap/oss/phosphor/fosstars/nvd/NVD.java:214]: The `forEach` applies `collector` to the NVD entries.

File: src/main/java/com/sap/oss/phosphor/fosstars/nvd/data/CpeMatch.java (hunk 2 above)
Comment (artem-smotrakov) [reply src/main/java/com/sap/oss/phosphor/fosstars/nvd/data/CpeMatch.java:18]: Yes, `JsonPropertyOrder` should contains "including" ones.

File: src/main/java/com/sap/oss/phosphor/fosstars/nvd/data/CpeMatch.java (hunk 3 above)
Comment (artem-smotrakov) [reply src/main/java/com/sap/oss/phosphor/fosstars/nvd/data/CpeMatch.java:28]: Yes, `JsonIgnoreProperties` should contain "excluding" ones.

**Comment by artem-smotrakov** [comment]
Thanks for the review @sourabhsparkala ! I've addressed your comments.

**Review by sourabhsparkala — APPROVED** [review]

**Pull request merged**